packages/smooth_app/lib/pages/prices/price_proof_page.dart

```diff
@@ -119,11 +166,132 @@ class _PriceProofPageState extends State<PriceProofPage> {
       uriHelper: ProductQuery.uriPricesHelper,
     );
     if (prices.isError) {
+      setState(() {
+        _isLoading = false;
+      });
       return;
     }
-    _existingPrices = prices.value.items ?? <Price>[];
-    if (mounted) {
-      setState(() {});
+    setState(() {
+      _existingPrices = prices.value.items ?? <Price>[];
+      _isLoading = false;
+    });
+  }
+}
+
+/// Widget to display a grid of proofs with price count badges
+class ProofGridItem extends StatefulWidget {
+  const ProofGridItem({
+    super.key,
+    required this.proof,
+    required this.onTap,
+  });
+
+  final Proof proof;
+  final VoidCallback onTap;
+
+  @override
+  State<ProofGridItem> createState() => _ProofGridItemState();
+}
+
+class _ProofGridItemState extends State<ProofGridItem> {
+  List<Price>? _prices;
+  bool _isLoading = true;
+
+  @override
+  void initState() {
+    super.initState();
+    unawaited(_loadPrices());
+  }
+
+  Future<void> _loadPrices() async {
+    if (PriceModel.isProofNotGoodEnough(widget.proof)) {
+      setState(() {
```

Comment:
For the record you're supposed to check `if(mounted)` before `setState` in an async code.